File: src/main/java/com/neo/caption/ocr/service/FileService.java
package com.neo.caption.ocr.service;

import com.neo.caption.ocr.domain.dto.FileChecksumDto;
import com.neo.caption.ocr.domain.dto.FileChunkDto;

import java.io.File;
import java.nio.file.Path;

public interface FileService {

    <T> void saveJsonToFile(Path path, T object);

    <T> T readJsonFromFile(Path path, Class<T> tClass);

    String getFileHeader(File file);

    void createTempDirectory(String hash);

    void uploadFileChunk(FileChunkDto fileChunkDto);

    void combineFileChunk(FileChecksumDto fileChecksumDto);

    Path getHashFile(String hash);

    void removeHashCache(String hash);
}

File: src/main/kotlin/com/neo/caption/ocr/service/FileService.kt
package com.neo.caption.ocr.service

import com.neo.caption.ocr.domain.entity.FileChecksum
import com.neo.caption.ocr.domain.entity.FileChunk
import com.neo.caption.ocr.domain.vo.SavedFileVo
import org.opencv.videoio.VideoCapture

interface FileService {

    fun createWorkingDirectory(): String

    fun saveFileChunk(fileChunk: FileChunk): SavedFileVo

    fun combineFileChunk(fileChecksum: FileChecksum): SavedFileVo

    fun openVideoFile(projectId: String): VideoCapture

}